releases/release-4.0.16.md

```diff
@@ -0,0 +1,106 @@
+---
+title: TiDB 4.0.16 Release Notes
+---
+
+# TiDB 4.0.16 Release Notes
+
+Release Date: December 10, 2021
+
+TiDB version: 4.0.16
+
+## Compatibility changes
+
++ TiKV
+
+    - When casting invalid utf8 string to real, try to truncate the utf8 prefix instead of returning error [#11466](https://github.com/tikv/tikv/issues/11466)
```

Comment:
```suggestion
    - When casting invalid utf8 string to real, try to truncate the utf8 prefix instead of returning an error [#11466](https://github.com/tikv/tikv/issues/11466)
```